tests/hil/scripts/pytest-hil/bmpboard.py

```diff
@@ -0,0 +1,24 @@
+from zephyrboard import ZephyrBoard
+import subprocess
+from time import sleep
+
+class BmpBoard(ZephyrBoard):
+    def program(self, fw_image):
+        command = ['gdb',
+                   '-ex', "set confirm off",
+                   '-ex', "target extended-remote {}".format(self.bmp_port),
+                   '-ex', "monitor swdp_scan",
+                   '-ex', "attach 1",
+                   '-ex', f"load {fw_image}",
+                   '-ex', "kill",
+                   '-ex', "quit"
+                   ]
+
+        subprocess.run(command)
+
+        # Arbitrary sleep while the software-based USB serial becomes available
+        sleep(3)
+
+    @property
+    def USES_WIFI(self):
```

Comment:
Rather than assume this for all boards that use the black magic probe, we should have an additional class for the RAK5010 that inherits from both `ZephyrBoard` and `BmpBoard`. See the 1024 for an example (inheriting from `ZephyrBoard` and `JLinkBoard`).

Reply:
Okay, so maybe let's move back to `RAK5010` and only introduce `BMPBoard` (or `BmpBoard`) once needed for other platforms. `sleep(3)` above in `__init__()` is also RAK5010 specific.

Reply:
I'm ok either way, I think it could be more readable to add the additional class (to make it clear what's bmp-specific and what's RAK5010 specific). We can add the delay by calling the super function:

```
def program(self, *args, **kwargs):
    BMPBoard.program(self, *args, **kwargs)
    
    sleep(3)
```

Reply:
Good suggestions, I've broken this out into the separate programmer and board files.